core/new-gui/src/app/dashboard/component/feature-container/user-project-section/project-page/ngbd-modal-add-project-workflow/ngbd-modal-add-project-workflow.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import { Observable, forkJoin } from "rxjs";
+import { WorkflowPersistService } from "src/app/common/service/workflow-persist/workflow-persist.service";
+import { DashboardWorkflowEntry } from "src/app/dashboard/type/dashboard-workflow-entry";
+import { UserProjectService } from "src/app/dashboard/service/user-project/user-project.service";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-add-project-workflow-modal",
+  templateUrl: "./ngbd-modal-add-project-workflow.component.html",
+  styleUrls: ["./ngbd-modal-add-project-workflow.component.scss"]
+})
+export class NgbdModalAddProjectWorkflowComponent implements OnInit {
+  @Input() addedWorkflows!: DashboardWorkflowEntry[];
+  @Input() projectId!: number;
+
+  public unaddedWorkflows: DashboardWorkflowEntry[] = [];
+  public checkedWorkflows: boolean[] = [];
+  private addedWorkflowKeys: Set<number> = new Set<number>();
+
+  constructor(
+    public activeModal: NgbActiveModal,
+    private workflowPersistService: WorkflowPersistService,
+    private userProjectService: UserProjectService
+  ) {}
+
+  ngOnInit(): void {
+    this.addedWorkflows.forEach(workflowEntry => this.addedWorkflowKeys.add(workflowEntry.workflow.wid!)); // TODO : would force casting cause any issues?
+    this.refreshProjectWorkflowEntries();
```

Comment:
it seems that the `addedWorkflows` is a local cache between the life cycles of `ngbd-modal`s. I suggest we remove all kinds of local caches and always fetch backend for the latest information. the reason is in share editing context, there could be more than one frontend editing the same project. using a local cache of any information would make the synchronization harder.

Reply:
discussed usage of cache and resolved potential issues with synchronization